packages/diagrams/frontend/sirius-components-diagrams/src/renderer/layout/useDistributeElements.ts

```diff
@@ -0,0 +1,398 @@
+          const currentSelectedNode = selectedNodes[index];
+          const previousNode = selectedNodes[index - 1];
+
+          let newPosition: number = direction === 'horizontal' ? node.position.x : node.position.y;
+
+          if (index > 0 && index < selectedNodes.length - 1 && previousNode && currentSelectedNode) {
+            newPosition =
+              direction === 'horizontal'
+                ? previousNode.position.x + (previousNode.width ?? 0) + gap
+                : previousNode.position.y + (previousNode.height ?? 0) + gap;
+            currentSelectedNode.position[direction === 'horizontal' ? 'x' : 'y'] = newPosition;
+          }
+
+          return {
+            ...node,
+            position: {
+              ...node.position,
+              [direction === 'horizontal' ? 'x' : 'y']: newPosition,
+            },
+          };
+        });
+
+        setNodes(updatedNodes);
+        const finalDiagram: RawDiagram = {
+          nodes: [...updatedNodes] as Node<NodeData, DiagramNodeType>[],
+          edges: getEdges(),
+        };
+        synchronizeLayoutData(refreshEventPayloadId, finalDiagram);
+      }
+    }, []);
+  };
+
+  const distributeAlign = (orientation: 'left' | 'right' | 'top' | 'bottom' | 'center' | 'middle') => {
+    return useCallback((selectedNodeIds: string[], refElementId: string | null) => {
+      processLayoutTool(
+        selectedNodeIds,
+        (selectedNodes, refNode) => {
+          selectedNodes.sort(getComparePositionFn('horizontal'));
+          return getNodes().map((node) => {
+            if (!selectedNodeIds.includes(node.id)) {
+              return node;
+            }
+            const referencePositionValue: number = (() => {
+              switch (orientation) {
+                case 'left':
+                  return refNode.position.x;
+                case 'right':
+                  return refNode.position.x + (refNode.width ?? 0) - (node.width ?? 0);
+                case 'center':
+                  return refNode.position.x + (refNode.width ?? 0) / 2 - (node.width ?? 0) / 2;
+                case 'top':
+                  return refNode.position.y;
+                case 'bottom':
+                  return refNode.position.y + (refNode.height ?? 0) - (node.height ?? 0);
+                case 'middle':
+                  return refNode.position.y + (refNode.height ?? 0) / 2 - (node.height ?? 0) / 2;
+                default:
+                  return 0;
+              }
+            })();
+
+            const referencePositionVariable: string = (() => {
+              switch (orientation) {
+                case 'left':
+                case 'right':
+                case 'center':
+                  return 'x';
+                case 'top':
+                case 'bottom':
+                case 'middle':
+                  return 'y';
+                default:
+                  return '';
```

Comment:
Should the default value be x or y?

Reply:
Since the orientation variable is correctly typed, I've removed the default case, all cases being covered